apps/hash-frontend/src/pages/shared/block-collection/shared/mention-suggester.tsx

```diff
@@ -171,7 +171,7 @@ export const MentionSuggester: FunctionComponent<MentionSuggesterProps> = ({
     fetchPolicy: "cache-and-network",
   });
 
-  const entitiesSubgraph = data?.structuralQueryEntities as
+  const entitiesSubgraph = data?.structuralQueryEntities.subgraph as
     | Subgraph<EntityRootType>
     | undefined;
```

Comment:
We should probably get rid of the numerous casts to `Subgraph<EntityRootType>` and make use of some `isEntityRootedSubgraph` type assertion instead, that way we can type the input as `Subgraph` and reference entities in `Subgraph<EntityRootType>` more reliably.

This would've also prevented us from forgetting to update this here.

Reply:
It's also possible that there may be more we haven't caught yet, so it might be worth doing a sweep of all casts either way.

Reply:
Offline resolution: we'll do both of the things suggested in these PR comments.

Reply:
I've replaced the casts with assertions in https://github.com/hashintel/hash/pull/3462/commits/6c50f15184b22dcb40531bdeebe1e2cc69df536c